### DISCUSSION TITLE
Failed to build oqs-chromium

### DISCUSSION THREAD

**Discussion by KrishCrS** [open-quantum-safe/oqs-demos#239] [Q&A] [answered]
Hi, 
I stuck at "git apply oqs-changes.patch".
I followed all of the instructions from [oqs-demos/chromium](https://github.com/open-quantum-safe/oqs-demos/blob/main/chromium/README-Linux.md). When I ran "git apply oqs-changes.patch" at [step 5](https://github.com/open-quantum-safe/oqs-demos/blob/main/chromium/README-Linux.md#5-enable-quantum-safe-crypto), it failed with the following error:
error: patch failed: net/cert/pki/signature_algorithm.h:44
error: net/cert/pki/signature_algorithm.h: patch does not apply
error: patch failed: third_party/boringssl/BUILD.gn:48
error: third_party/boringssl/BUILD.gn: patch does not apply

Ubuntu 22.04.3LTS is the version I'm using. Any assistance is highly appreciated.
![oqs-changes_patch](https://github.com/open-quantum-safe/oqs-demos/assets/142003662/43ac2bc0-d987-43e2-ba67-13b5bc345811)

**Comment by baentsch** [comment]
Tagging @Raytonne @Jijun

**Comment by Raytonne** [comment]
Hello - may I know what's the Chromium version? That instruction should work for Chromium `117.0.5863.0`
For other Chromium versions, you may need to make some edits based on the patch by yourself

**Reply by KrishCrS** [reply]
I followed these instructions to get the source code of chromium and additonal build dependencies and some additional binaries :

1. $ git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git
2. $ export PATH="$PATH:${PWD}/depot_tools"
3. $ mkdir chromium && cd chromium
4. $ fetch --nohooks chromium
5. $ cd src
6. $ ./build/install-build-deps.sh
7. $ gclient runhooks

Can you please suggest me how to download a specific version of chromium source code ?

**Reply by Raytonne** [reply, marked as answer]
```
git checkout tags/117.0.5863.0
gclient sync
```
should fetch `117.0.5863.0`.

**Reply by KrishCrS** [reply]
Thanks a lot.
Another query: as liboqs 0.8.0 is used in your case, and as the current liboqs version is 0.9.0, will cloning the git repo work perfectly, or should we have to download and install the source code of liboqs 0.8.0?

**Reply by Raytonne** [reply]
0.9 may work, you can try it. If you don’t want to try, then use 0.8

**Reply by KrishCrS** [reply]
Can you @Raytonne please suggest how to confirm whether Chromium has been properly enabled with OQS support? I've attempted to follow the **test** instructions provided in the [README-Linux.md](https://github.com/open-quantum-safe/oqs-demos/commit/0b5a94fab633b1a428641de49f11f8b9031c982b#diff-9cb5f72748c8338ff4c54d34e9e18739c895eec58f7f8e91ee3342558e513b3b) onwards line number 46, but unfortunately, neither method appears to be effective.

**Comment by baentsch** [comment]
> ```
> git checkout tags/117.0.5863.0
> gclient sync
> ```
> 
> should fetch `117.0.5863.0`.

Would it be sensible to add this to the documentation to avoid future questions regarding this matter? Same regarding an advice on `liboqs` versions? In general, we strive to have all oqs-demos integrations run with the latest `liboqs` (hence, typically have CI test against `liboqs` "main") -- but then again, there is no CI for `chromium`, so some documentation statement may help other users.

**Comment by baentsch** [comment]
Resolved via #240. Thanks @Raytonne !
